src/libs/blueprint/blueprint_mesh.cpp

```diff
@@ -94,8 +133,504 @@ bool
+    
+    // required: "coordsets", with at least one child
+    //  each child must conform to "mesh::coordset"
+    if(!n.has_child("coordsets"))
+    {
+        info["errors"].append().set("missing child node \"coordsets\"");
+        res = false;
+    }
+    else if(n["coordsets"].number_of_children() == 0)
+    {
+        info["errors"].append().set("\"coordsets\" has no children");
+        res = false;
+    }
+    else
+    {
+        NodeConstIterator itr = n["coordsets"].children();
+    
+        while(itr.has_next())
+        {
+            const Node &chld = itr.next();
+            const std::string chld_name = itr.path();
+
+            if(!coordset::verify(chld,info["coordsets/" + chld_name]))
+            {
+                info["errors"].append().set(chld_name 
+                                + " is not a valid blueprint::mesh::coordset");
+                res = false;
+            }
+        }
+    }
+    
+    // required: "topologies",  with at least one child
+    // each child must conform to "mesh::topology"
+    if(!n.has_child("topologies"))
+    {
+        info["errors"].append().set("missing child node \"topologies\"");
+        res = false;
+    }
+    else if(n["topologies"].number_of_children() == 0)
+    {
+        info["errors"].append().set("\"topologies\" has no children");
+        res = false;
+    }
+    else
+    {
+    
+        NodeConstIterator itr = n["topologies"].children();
+        
+        while(itr.has_next())
+        {
+            const Node &chld = itr.next();
+            const std::string chld_name = itr.path();
+
+            if(!topology::verify(chld,info["topologies"][chld_name]))
+            {
+                info["errors"].append().set(chld_name 
+                                + " is not a valid blueprint::mesh::topology");
+                res = false;
+            }
+
+            // make sure the topology references a valid coordset
+            std::string coords_name = chld["coordset"].as_string();
+            
+            if(!n["coordsets"].has_child(coords_name))
+            {
+                std::ostringstream oss;
+                oss << "topology "
+                    << "\"" << chld_name  << "\" "
+                    << "references a non-existent coordset "
+                    << "\"" << coords_name  << "\" ";
+                info["errors"].append().set(oss.str());
+                res = false;
+            }
+            else if(info["coordsets"][coords_name]["valid"].as_string() != "true")
+            {
+                std::ostringstream oss;
+                oss << "topology "
+                    << "\"" << chld_name  << "\" "
+                    << "references a invalid coordset "
+                    << "\"" << coords_name  << "\" ";
+                info["errors"].append().set(oss.str());
+                res = false;
+            }
+        }
+    }
+    
+    // optional: "fields", each child must conform to "mesh::field"
```

Comment:
It may be worth noting in the `info` node that this section is optional as is done with the extra dimensions for nodes in the `coordset::verify_coordset_uniform` function below.